include/swift/AST/DiagnosticGroups.def

```diff
@@ -78,6 +78,7 @@ GROUP(TrailingClosureMatching, "trailing-closure-matching")
 GROUP(UnknownWarningGroup, "unknown-warning-group")
 GROUP(CompilationCaching, "compilation-caching")
 GROUP(WeakMutability, "weak-mutability")
+GROUP(PerfHintReturnTypeImplicitCopy, "perf-hint-return-type-implicit-copy")
```

Comment:
This is... very specific. Can we perhaps drop the "PerfHint"/"perf-hint-" prefix here and put them within a "PerformanceHints" group?

Reply:
The goal here was to allow each check/hint to be enabled/disabled _individually_. And the idea is the each hint will actually have a small handful of diagnostic varieties, functions and closures being separate diagnostic entires in this PR being an example.